carbon_black_cloud/plugin.spec.yaml

```diff
@@ -0,0 +1,456 @@
+      title: "Rooted by Sensor Time"
+      type: string
+      description: "Rooted by Sensor Time"
+      required: false
+    scanLastActionTime:
+      title: "Scan Last Action Time"
+      type: integer
+      description: "Scan last action time"
+      required: false
+    scanLastCompleteTime:
+      title: "Scan Last Complete Time"
+      type: integer
+      description: "Scan last complete time"
+      required: false
+    scanStatus:
+      title: "Scan Status"
+      type: string
+      description: "Scan status"
+      required: false
+    sensorKitType:
+      title: "Sensor Kit Type"
+      type: string
+      description: "Sensor kit type"
+      required: false
+    sensorOutOfDate:
+      title: "Sensor Out of Date"
+      type: boolean
+      description: "Sensor out of date"
+      required: false
+    sensorPendingUpdate:
+      title: "Sensor Pending Update"
+      type: boolean
+      description: "Sensor pending update"
+      required: false
+    sensorStates:
+      title: "Sensor States"
+      type: "[]string"
+      description: "Sensor states"
+      required: false
+    sensorVersion:
+      title: "Sensor Version"
+      type: string
+      description: "Sensor version"
+      required: false
+    status:
+      title: "Status"
+      type: string
+      description: "Status"
+      required: false
+    targetPriorityType:
+      title: "Target Priority Type"
+      type: string
+      description: "Target priority type"
+      required: false
+    testId:
+      title: "Test ID"
+      type: integer
+      description: "Test ID"
+      required: false
+    uninstallCode:
+      title: "Uninstall Code"
+      type: string
+      description: "Uninstall code"
+      required: false
+    uninstalledTime:
+      title: "Uninstalled Time"
+      type: string
+      description: "Uninstalled time"
+      required: false
+    vdiBaseDevice:
+      title: "VDI Base Device"
+      type: string
+      description: "VDI Base Device"
+      required: false
+    virtualMachine:
+      title: "Virtual Machine"
+      type: boolean
+      description: "Virtual machine"
+      required: false
+    virtualizationProvider:
+      title: "Virtualization Provider"
+      type: string
+      description: "Virtualization Provider"
+      required: false
+    windowsPlatform:
+      title: "Windows Platform"
+      type: string
+      description: "Windows platform"
+      required: false
+
+connection:
+  api_id:
+    title: API ID
+    type: string
+    description: API ID
+    example: ABCDE12345
+    required: true
+    order: 1
+  api_secret_key:
+    title: API Secret Key
+    type: credential_secret_key
+    description: API Secret Key
+    example: ABCDEFG12345ABCD1234ABCD
+    required: true
+    order: 2
+  url:
+    title: URL
+    type: string
+    description: API URL
+    default: https://api5.conferdeploy.net/integrationServices/v3/
+    example: https://api5.conferdeploy.net/integrationServices/v3/
```

Comment:
I think this is a partner URL and not the actual URL a customer would use.

We'll need to reach out to vendor to confirm what it is, if it's not listed in the documentation

Reply:
We should also not make the customer pass a full URL, and have code in the connection to handle if they do.

E.g. /integrationServices/v3/ - should just allow `https://api5.conferdeploy.net/`

and if they paste the wrong thing we should just trim it in the code. Recently had this happen in the SentinelOne plugin for a user.

Reply:
https://developer.carbonblack.com/reference/carbon-black-cloud/authentication/

That's why I did what I did there...the URL configuration sucks.